Revert pv level indexing change

lbdreyer opened the pull request:
Closes #150 

I have also convinced myself that the indexing before was correct, based on looking at some data produced from ECMWF, rather than our own sample data

@pp-mo Are you able to review?

lbdreyer commented on iris_grib/_load_convert.py:
```diff
@@ -1492,14 +1492,14 @@ def hybrid_factories(section, metadata):
             metadata['aux_coords_and_dims'].append((coord, None))
             # Create the sigma scalar coordinate.
-            offset = NV // 2 + scaledValue - 1
+            offset = NV // 2 + scaledValue
```
Note! prior to #106 this would have been `scaledValue + NV // 2` but I don't think that is correct

pp-mo commented on iris_grib/_save_rules.py:
```diff
@@ -953,9 +953,9 @@ def set_fixed_surfaces(cube, grib, full3d_cube=None):
-        # values are indexed according to model-level-number,
+        # value *before* each set of n-levels coefficients.
+        # Note: values are indexed according to model_level_number, using
+        # zero-based numbering,
```
I think the 'using zero-based numbering' bit is still confusing!
You might add to the following ..
`# I.E. sigma, delta = PV[i], PV[NV/2+i]  : where i=1..n_levels`

pp-mo commented on iris_grib/_load_convert.py:
```diff
@@ -1510,14 +1510,14 @@ def hybrid_factories(section, metadata):
+            # Create the level height/pressure scalar coordinate.
             pv = section['pv']
-            offset = scaledValue - 1
+            offset = scaledValue
```
Maybe worth an extra comment here also, about 'scaledValue' being 1-based, or "1..n_levels".
And/or maybe rename the variable 'level_number' ??

lbdreyer replied:
I don't agree with renaming the variable `level_number` as offset is meant to be the offset within the the pv array, see line 1520 where the offset is different

pp-mo replied:
Sorry, I meant that you could rename ***`scaledValue`*** as `level_number`.  I.E. what you get as the principal coordinate value is in this case a level number, i.e. a 1..N value.

I will check out your other changes..

pp-mo commented:
Thanks @lbdreyer 
Function is good.  Just a couple of ideas for making code a bit clearer.
Then will merge !

lbdreyer commented:
@pp-mo I have addressed your comments in the most recent commit. 
I had a little difficulty coming up with good commits so let me know if you can think of a better way to explain it.

pp-mo commented:
> let me know if you can think of a better way to explain it.

No, this does it for me, I think that _is_ clear.

pp-mo commented:
Thanks @lbdreyer !
